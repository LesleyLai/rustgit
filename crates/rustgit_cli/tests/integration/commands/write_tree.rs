use crate::common::{git, populate_folder, rustgit};
use assert_cmd::prelude::*;
use std::str::from_utf8;
use test_utils::test_path;

#[test]
fn write_tree() -> anyhow::Result<()> {
    let working_dir = test_path!();

    git(&working_dir).init();

    populate_folder(&working_dir);

    git(&working_dir).stage(".");

    let tree_hash = rustgit(&working_dir).write_tree()?;

    let command = git(&working_dir)
        .args(["ls-tree", &tree_hash])
        .assert()
        .success();
    let output = from_utf8(&command.get_output().stdout)?;
    insta::assert_snapshot!(output);

    Ok(())
}
