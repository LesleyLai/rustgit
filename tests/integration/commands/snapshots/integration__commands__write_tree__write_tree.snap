---
source: tests/integration/commands/write_tree.rs
assertion_line: 22
expression: output
---
040000 tree 91e1483644d087af54a6e8aac15a08c482bb9fb1	dir1
040000 tree cf8e933fedbe540f9881ba4dc34b034785834227	dir2
100644 blob b6fc4c620b67d95f953a5c1c1230aaab5db5a1b0	file1.txt
